scripts/biomedical/VirtualMetabolicHuman/format_metabolite.py

```diff
@@ -0,0 +1,346 @@
+    pre-existing matching hmdb ids on dc
+    Args:
+        arr_hmdb = array with hmdb ids to query
+    Returns:
+        result = result of dc query
+    """
+    query_str = """
+    SELECT DISTINCT ?drug ?id
+    WHERE {{
+    ?drug typeOf ChemicalCompound .
+    ?drug humanMetabolomeDatabaseID ?id .
+    ?drug humanMetabolomeDatabaseID {0} .
+    }}
+    """.format(arr_hmdb)
+    result = dc.query(query_str)
+    return result
+
+
+def kegg_query(arr_kegg):
+    """
+    Queries dc using the python api, to find
+    if the elements of the input list, have a 
+    pre-existing matching kegg ids on dc
+    Args:
+        arr_hmdb = array with kegg ids to query
+    Returns:
+        result = result of dc query
+    """
+    query_str = """
+    SELECT DISTINCT ?drug ?id
+    WHERE {{
+    ?drug typeOf ChemicalCompound .
+    ?drug keggCompoundID ?id .
+    ?drug keggCompoundID {0} .
+    }}
+    """.format(arr_kegg)
+    result = dc.query(query_str)
+    return result
+
+
+def chebi_query(arr_chebi):
+    """
+    Queries dc using the python api, to find
+    if the elements of the input list, have a 
+    pre-existing matching chebi ids on dc
+    Args:
+        arr_hmdb = array with chebi ids to query
+    Returns:
+        result = result of dc query
+    """
+    query_str = """
+    SELECT DISTINCT ?drug ?id
+    WHERE {{
+    ?drug typeOf ChemicalCompound .
+    ?drug chebiID ?id .
+    ?drug chebiID {0} .
+    }}
+    """.format(arr_chebi)
+    result = dc.query(query_str)
+    return result
+
+
+def drugbank_query(arr_drug):
+    """
+    Queries dc using the python api, to find
+    if the elements of the input list, have a 
+    pre-existing matching drugbank ids on dc
+    Args:
+        arr_hmdb = array with drugbank ids to query
+    Returns:
+        result = result of dc query
+    """
+    query_str = """
+    SELECT DISTINCT ?drug ?id
+    WHERE {{
+    ?drug typeOf ChemicalCompound .
+    ?drug drugBankMetaboliteID ?id .
+    ?drug drugBankMetaboliteID {0} .
+    }}
+    """.format(arr_drug)
+    result = dc.query(query_str)
+    return result
+
+
+def shard(list_to_shard, shard_size):
+    """
+    Breaks down a list into smaller 
+    sublists, converts it into an array
+    and appends the array to the master
+    list
+    Args:
+        list_to_shard = original list
+        shard_size = size of subist
+    Returns:
+        sharded_list = master list with
+        smaller sublists 
+    """
+    sharded_list = []
+    for i in range(0, len(list_to_shard), shard_size):
+        shard = list_to_shard[i, i + shard_size]
+        arr = np.array(shard)
+        sharded_list.append(arr)
+    return sharded_list
+
+
+def main():
+    file_input = sys.argv[1]
+    file_output = sys.argv[2]
+    file_hmdb = sys.argv[3]
+
+    df = pd.read_csv(file_input, sep='\t')
+    dfh = pd.read_csv('hmdb.csv')
+
+    #inchikey matches with dc
+    list_inchi = df[['inchiKey']].T.stack().tolist()
```

Comment:
Similarly here, you'll note that these blocks of code are very similar (only the col string is different). This is a good candidate to pull out into a helper that can be reused, and every time that's done - it's less code, fewer bugs, easier to maintain and extend.